### PR TITLE
Add Clang assertions trunk for OpenCL C and C++ for OpenCL

### PR DESCRIPTION
**Pull request by jzzheng22**
This PR adds Clang built with assertions enabled for OpenCL C and C++ for OpenCL.

### CODE REVIEW

File: etc/config/openclc.amazon.properties [hunk 1]
```diff
@@ -117,6 +117,14 @@ compiler.armv7-oclcclang-trunk.semver=(trunk)
 # Arm v7-a with Neon and VFPv3
 compiler.armv7-oclcclang-trunk.options=-target arm-linux-gnueabi --gcc-toolchain=/opt/compiler-explorer/arm/gcc-8.2.0/arm-unknown-linux-gnueabi --sysroot=/opt/compiler-explorer/arm/gcc-8.2.0/arm-unknown-linux-gnueabi/arm-unknown-linux-gnueabi/sysroot
 
+compiler.armv7-oclcclang-trunk-assertions.name=armv7-a clang (trunk, assertions)
+compiler.armv7-oclcclang-trunk-assertions.exe=/opt/compiler-explorer/clang-assertions-trunk/bin/clang
+compiler.armv7-oclcclang-trunk-assertions.demangler=/opt/compiler-explorer/gcc-snapshot/bin/c++filt
+compiler.armv7-oclcclang-trunk-assertions.objdumper=/opt/compiler-explorer/gcc-snapshot/bin/objdump
+compiler.armv7-oclcclang-trunk-assertions.semver=(assertions trunk)
+# Arm v7-a with Neon and VFPv3
+compiler.armv7-oclcclang-trunk.options=-target arm-linux-gnueabi --gcc-toolchain=/opt/compiler-explorer/arm/gcc-8.2.0/arm-unknown-linux-gnueabi --sysroot=/opt/compiler-explorer/arm/gcc-8.2.0/arm-unknown-linux-gnueabi/arm-unknown-linux-gnueabi/sysroot
```
Comment (mattgodbolt) [inline etc/config/openclc.amazon.properties:126]: is this missing the `-assertions` suffix on the compiler name?

File: etc/config/openclc.amazon.properties [hunk 2]
```diff
@@ -125,6 +133,14 @@ compiler.armv8-oclcclang-trunk.semver=(trunk)
 # Arm v8-a
 compiler.armv8-oclcclang-trunk.options=-target aarch64-linux-gnu --gcc-toolchain=/opt/compiler-explorer/arm64/gcc-8.2.0/aarch64-unknown-linux-gnu --sysroot=/opt/compiler-explorer/arm64/gcc-8.2.0/aarch64-unknown-linux-gnu/aarch64-unknown-linux-gnu/sysroot
 
+compiler.armv8-oclcclang-trunk-assertions.name=armv8-a clang (trunk, assertions)
+compiler.armv8-oclcclang-trunk-assertions.exe=/opt/compiler-explorer/clang-assertions-trunk/bin/clang
+compiler.armv8-oclcclang-trunk-assertions.demangler=/opt/compiler-explorer/gcc-snapshot/bin/c++filt
+compiler.armv8-oclcclang-trunk-assertions.objdumper=/opt/compiler-explorer/gcc-snapshot/bin/objdump
+compiler.armv8-oclcclang-trunk-assertions.semver=(assertions trunk)
+# Arm v8-a
+compiler.armv8-oclcclang-trunk.options=-target aarch64-linux-gnu --gcc-toolchain=/opt/compiler-explorer/arm64/gcc-8.2.0/aarch64-unknown-linux-gnu --sysroot=/opt/compiler-explorer/arm64/gcc-8.2.0/aarch64-unknown-linux-gnu/aarch64-unknown-linux-gnu/sysroot
```
Comment (mattgodbolt) [inline etc/config/openclc.amazon.properties:142]: ditto?

**Review by mattgodbolt — CHANGES_REQUESTED** [review]
Couple super minor typos but otherwise seems OK!

File: etc/config/openclc.amazon.properties (hunk 1 above)
Comment (jzzheng22) [reply etc/config/openclc.amazon.properties]: Thanks! That's what I get for copy pasting without double checking properly

**Comment by jzzheng22** [comment]
Hi! Just wanted to ping this - I've fixed the typos :)

**Review by mattgodbolt — APPROVED** [review]
Thanks! Sorry for our unwieldy configuration format!

**Pull request merged**